traversal/selector/exploreInterpretAs.go

```diff
@@ -0,0 +1,65 @@
+package selector
+
+import (
+	"fmt"
+
+	"github.com/ipld/go-ipld-prime/datamodel"
+)
+
+type ExploreInterpretAs struct {
+	next Selector // selector for element we're interested in
+	adl  string   // reifier for the ADL we're interested in
+}
+
+// Interests for ExploreIndex is just the index specified by the selector node
+func (s ExploreInterpretAs) Interests() []datamodel.PathSegment {
+	return s.next.Interests()
+}
+
+// Explore returns the node's selector if
+// the path matches the index for this selector or nil if not
+func (s ExploreInterpretAs) Explore(n datamodel.Node, p datamodel.PathSegment) (Selector, error) {
+	return s.next, nil
+}
+
+// Decide always returns false because this is not a matcher
+func (s ExploreInterpretAs) Decide(n datamodel.Node) bool {
+	return false
+}
+
+// Transform indicates how this selector expects to transform the current datamodel.Node.
+func (s ExploreInterpretAs) Transform() string {
+	return s.adl
+}
+
+// Transformable provides for a feature detection interface on selectors to understand when
+// and if Transformation of the datamodel.node should be attempted when performing traversals.
+type Transformable interface {
+	Transform() string
```

Comment:
I would have expected this method name to contain "reifier" or "ADL"; it's not really a generic "transform" given that it only supports whatever is registered via a string registry.

What do we call those strings? If, for example, those are "named reifiers", then I'd perhaps go with:

```
type Reifiable interface {
    NamedReifier() string
}
```

Reply:
Alternatively, if your argument is that this supports altering nodes in any way, not just reification, then perhaps NamedTransform.

Reply:
I think the use of transform here was to not overload the current `Reify` so that the two uses were named distinctly and there wasn't confusion in talking about them. I'll switch to your suggested naming though, since in this design it really is the same concept/operation